src/extremeweatherbench/evaluate.py

```diff
@@ -194,25 +192,112 @@ def _evaluate_case(
             for metric in individual_case.metrics_list:
                 metric_instance = metric()
                 logging.debug(
-                    "metric %s computing for %s", metric_instance.name, data_var
+                    "gridded metric %s computing for %s", metric_instance.name, data_var
                 )
                 result = metric_instance.compute(forecast_da, gridded_obs_da)
-                case_results[data_var][metric_instance.name] = result
+                case_results["gridded"][data_var][metric_instance.name] = result
+                logger.debug(
+                    "gridded, %s, %s, %s", data_var, metric_instance.name, result
+                )
     if point_obs is not None:
-        raise NotImplementedError("Point obs evaluation not implemented as of 0.1.0")
+        case_results["point"] = {}
+        case_subset_point_obs = point_obs.loc[point_obs["id"] == individual_case.id]
+        case_subset_point_obs = case_subset_point_obs.rename(columns=utils.ISD_MAPPING)
+        case_subset_point_obs["longitude"] = utils.convert_longitude_to_360(
+            case_subset_point_obs["longitude"]
+        )
+        case_subset_point_obs = utils.unit_check(case_subset_point_obs)
+        case_subset_point_obs = case_subset_point_obs[
+            (
+                case_subset_point_obs["latitude"]
+                >= spatiotemporal_subset_forecast_ds["latitude"].min().values
+            )
+            & (
+                case_subset_point_obs["latitude"]
+                <= spatiotemporal_subset_forecast_ds["latitude"].max().values
+            )
+            & (
+                case_subset_point_obs["longitude"]
+                >= spatiotemporal_subset_forecast_ds["longitude"].min().values
+            )
+            & (
+                case_subset_point_obs["longitude"]
+                <= spatiotemporal_subset_forecast_ds["longitude"].max().values
+            )
+        ]
+
+        case_subset_point_obs_ds = case_subset_point_obs.set_index(
+            ["time", "station"]
+        ).to_xarray()
+
+        case_subset_point_obs_ds = case_subset_point_obs_ds.set_coords(
+            ["latitude", "longitude", "name", "elev", "id", "call"]
+        )
+        # remove unnecessary time dimension from lat and lon
+        for coord in ["latitude", "longitude"]:
+            case_subset_point_obs_ds[coord] = case_subset_point_obs_ds[coord].isel(
+                time=0, drop=True
+            )
+
+        point_forecast_subset_ds = utils.pick_points(
+            spatiotemporal_subset_forecast_ds,
+            case_subset_point_obs.drop_duplicates(
+                subset=["latitude", "longitude", "station"]
+            )[["latitude", "longitude", "elev", "id", "name"]],
+            config=eval_config,
+            tree_name=individual_case.id,
+        )
+        point_forecast_subset_ds = point_forecast_subset_ds.rename({"point": "station"})
```

Comment:
This call is doing a tremendous amount of heavy lifting. You should add comments to explain what is happening; at a minimum, I think you need a paragraph-length comment preceding this block explaining the format of the data sources you're comparing and what the expected output will look like.